packages/@aws-cdk/aws-sagemaker/lib/endpoint.ts

```diff
@@ -0,0 +1,228 @@
+     */
+    readonly initialVariantWeight?: number;
+
+    /**
+     * Instance type of the production variant.
+     *
+     * @default ml.c4.xlarge instance type.
+     */
+    readonly instanceType?: ec2.InstanceType;
+
+    /**
+     * The model aligned to the production variant.
+     *
+     */
+    readonly model: IModel;
+
+    /**
+     * Name of the production variant.
+     *
+     * @default same name as the model name
+     */
+    readonly variantName?: string;
+}
+
+export enum AcceleratorType {
+
+    LARGE = 'ml.eia1.large ',
+
+    MEDIUM = 'ml.eia1.medium',
+
+    XLARGE = 'ml.eia1.xlarge',
+}
+
+/**
+ * Defines a SageMaker Endpoint and associated configuration resource.
+ *
+ * @experimental
+ */
+export class Endpoint extends Resource {
+
+    /**
+     * The name of the endpoint.
+     *
+     * @attribute
+     */
+    public readonly endpointName: string;
+
+    /**
+     * Name of the endpoint configuration.
+     *
+     * @attribute
+     */
+    public readonly endpointConfigName: string;
+
+    private readonly productionVariants: ProductionVariant[] = [];
+
+    constructor(scope: Construct, id: string, props: EndpointProps= {}) {
+        super(scope, id);
+
+        // check that the production variants are defined
+        if (props.productionVariants && props.productionVariants.length > 0) {
+            this.productionVariants = props.productionVariants;
+        }
+
+        // apply a name tag to the endpoint resource
+        this.node.applyAspect(new Tag(NAME_TAG, this.node.path));
+
+        // create the endpoint configuration resource
+        const endpointConfig = new CfnEndpointConfig(this, 'EndpointConfig', {
+            kmsKeyId: (props.kmsKey) ? props.kmsKey.keyArn : undefined,
+            endpointConfigName: (props.configName) ? props.configName : (props.endpointName) ? props.endpointName + "Config" : undefined,
+            productionVariants: Lazy.anyValue({ produce: () => this.renderProductionVariants(this.productionVariants) })
+        });
+        this.endpointConfigName = endpointConfig.attrEndpointConfigName;
+
+        // create the endpoint resource
+        const endpoint = new CfnEndpoint(this, 'Endpoint', {
+            endpointConfigName: this.endpointConfigName,
+            endpointName: (props.endpointName) ? props.endpointName : undefined,
+        });
+        this.endpointName = endpoint.attrEndpointName;
+    }
+
+    /**
+     * Add production variant to the endpoint configution.
+     *
+     * @param productionVariant: The production variant to add.
+     */
+    public addProductionVariant(productionVariant: ProductionVariant): void {
+        this.productionVariants.push(productionVariant);
+    }
+
+    protected validate(): string[] {
+        const result = super.validate();
+        // check we have at least one production variant
+        if (this.productionVariants.length === 0) {
+            result.push("Must have at least one Production Variant");
```

Comment:
So here's an interesting pattern that you can use here.

Since at least one ProductionVariant is required, have in props:

```ts
productionVariant: ProductionVariant; // required
extraProductionVariants?: ProductionVariant[]; // optional
```

This way, you communicate to the clients of this class, at compile time, that at least one variant has to be provided.